middle_end/flambda/terms/continuation_params_and_handler.rec.ml

```diff
@@ -66,7 +66,20 @@ let pattern_match t ~f =
   pattern_match t ~f:(fun params { handler; } ->
     f params ~handler)
 
+module Pattern_match_pair_error = struct
+  type t = Parameter_lists_have_different_lengths
+
+  let to_string = function
+    | Parameter_lists_have_different_lengths ->
+      "Parameter lists have different lengths"
+end
+
 let pattern_match_pair t1 t2 ~f =
-  pattern_match_pair t1 t2 ~f:(
-    fun params { handler = handler1; } { handler = handler2; } ->
-      f params ~handler1 ~handler2)
+  pattern_match t1 ~f:(fun params1 ~handler:_ ->
+    pattern_match t2 ~f:(fun params2 ~handler:_ ->
+      if List.compare_lengths params1 params2 = 0 then
```

Comment:
Please leave a CR saying that we should think about whether Name_abstraction.Make_list should do this check itself.